pkg/transform/transform.go

```diff
@@ -0,0 +1,377 @@
+		newDest = strings.ToLower(newDest)
+	}
+
+	if len(split) > 1 {
+		return []byte(newDest + "#" + split[1]), nil
+	}
+	return []byte(newDest), nil
+}
+
+func (r *relLinkTransformer) Close(mdformatter.SourceContext) error { return nil }
+
+type frontMatterTransformer struct {
+	localLinksStyle LinksStyle
+	c               *FrontMatterConfig
+
+	// Vars.
+	origin FrontMatterOrigin
+	target FrontMatterTarget
+}
+
+type FrontMatterOrigin struct {
+	Filename    string
+	FirstHeader string
+	LastMod     string
+}
+
+type FrontMatterTarget struct {
+	FileName string
+}
+
+func (f *frontMatterTransformer) TransformFrontMatter(ctx mdformatter.SourceContext, frontMatter map[string]interface{}) ([]byte, error) {
+	b := bytes.Buffer{}
+	if err := f.c._template.Execute(&b, struct {
+		Origin      FrontMatterOrigin
+		Target      FrontMatterTarget
+		FrontMatter map[string]interface{}
+	}{
+		Origin:      f.origin,
+		Target:      f.target,
+		FrontMatter: frontMatter,
+	}); err != nil {
+		return nil, err
+	}
+
+	m := map[string]interface{}{}
+	if err := yaml.Unmarshal(b.Bytes(), m); err != nil {
+		return nil, errors.Wrapf(err, "generated template for %v is not a valid yaml", ctx.Filepath)
+	}
+
+	if f.localLinksStyle == Hugo {
+		if _, ok := m["slug"]; !ok {
+			m["slug"] = f.target.FileName
+		}
+	}
+
+	return mdformatter.FormatFrontMatter(m)
+}
+
+func (f *frontMatterTransformer) Close(mdformatter.SourceContext) error { return nil }
+
+func firstMatch(absRelPath string, trs []*TransformationConfig) (*TransformationConfig, bool) {
+	for _, tr := range trs {
+		if tr._glob.Match(absRelPath) {
+			return tr, true
+		}
+	}
+	return nil, false
+}
+
+func newTargetAbsRelPath(absRelPath string, tr *TransformationConfig) string {
+	if tr.Path == "" {
+		return absRelPath
+	}
+
+	if filepath.IsAbs(tr.Path) {
+		return tr.Path
+	}
+
+	return filepath.Join(filepath.Dir(absRelPath), tr.Path)
+}
+
+func copyFiles(src, dst string) (err error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return errors.Wrap(err, "cpy source")
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return errors.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return errors.Wrap(err, "cpy source")
+	}
+	defer errcapture.Close(&err, source, "src close")
+
+	if err := os.MkdirAll(filepath.Dir(dst), os.ModePerm); err != nil {
+		return err
+	}
+
+	destination, err := os.Create(dst)
+	if err != nil {
+		return errors.Wrap(err, "cpy dest")
+	}
+	defer errcapture.Close(&err, destination, "dst close")
+
+	_, err = io.Copy(destination, source)
+	return err
+}
+
+// TODO(bwplotka): Use formatter, remove the title etc.
+// Super hacky for now.
+func popFirstHeader(path string) (_ string, rest []byte, err error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", nil, err
+	}
+	defer errcapture.Close(&err, file, "close file")
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
Can we check `scanner.Err()` after this loop?

Reply:
Very good point, I missed this.

Where did you learn so many stuff about golang already??? ;p